supervisor/os/data_disk.py

```diff
@@ -190,3 +212,36 @@ async def migrate_disk(self, new_disk: str) -> None:
                 f"Can't restart device to finish disk migration: {err!s}",
                 _LOGGER.warning,
             ) from err
+
+    async def _format_device_with_single_partition(self, new_disk: Disk) -> None:
+        """Format device with a single partition to use as data disk."""
+        block_device: UDisks2Block = self.sys_dbus.udisks2.get_block_device(
+            new_disk.device_object_path
+        )
+
+        try:
+            await block_device.format(FormatType.GPT)
+        except DBusError as err:
+            capture_exception(err)
+            raise HassOSDataDiskError(
+                f"Could not format {new_disk.id}: {err!s}", _LOGGER.error
+            ) from err
+
+        await block_device.check_type()
+        if not block_device.partition_table:
+            raise HassOSDataDiskError(
+                "Block device does not contain a partition table after format, cannot create data partition",
+                _LOGGER.error,
+            )
+
+        try:
+            partition = await block_device.partition_table.create_partition(
+                0, 0, LINUX_DATA_PARTITION_GUID, EXTERNAL_DATA_DISK_PARTITION_NAME
+            )
+        except DBusError as err:
+            capture_exception(err)
+            raise HassOSDataDiskError(
+                f"Could not create new data partition: {err!s}"
+            ) from err
+
+        _LOGGER.info("New partition D-Bus object: %s", partition)
```

Comment:
Yea we can. Not like that though, partition is a string (object path of the new partition). Have to ask dbus for the value of its `org.freedesktop.UDisks2.Block.Device` property. Since that does add a small perf hit I'll add a log level check to see if its necessary before doing it.

Reply:
Oh wait, I guess I need the partition size anyway to do the other check. Ok nvm, this is no issue then